src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -811,7 +808,7 @@ keypairForSignature(Json::Value const& params, Json::Value& error)
             return {};
         }
 
-        if (secretType == jss::secret.c_str())
+        if (strcmp(secretType, jss::secret.c_str()) == 0) // don't assume that pointers are the same
```

Comment:
Please add a link in a comment somewhere to the bug report you filed.

Reply:
Thanks @thejohnfreeman, done.

Reply:
I meant in the code. When someone comes along and wonders, "why can't I assume the pointers are the same? that doesn't make any sense", it will be helpful to have an explanation right there.

Reply:
@thejohnfreeman added link in the code.